packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -487,7 +494,7 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         metered(_gasLimit)
     {
         (address token,) = gasPayingToken();
-        if (token != Constants.ETHER && msg.value != 0) {
+        if (isCustomGasToken() && msg.value != 0) {
```

Comment:
This is a good readability improvement but since it adds an extra call to the system config, i think we should revert this change. In the future we can consider an optimization that is:
- check transient storage for value
- if no value, call system config and populate transient storage
- return value from transient storage

This would basically be a local cache for the value

For now we should just take the readability hit and save the extra call but good thinking